x-pack/plugin/esql/qa/testFixtures/src/main/resources/semantic_text.csv-spec

```diff
@@ -173,3 +173,946 @@ host:keyword | semantic_text_field:semantic_text
+2           | "POINT(7580.93 2272.77)"
+3           | null
+;
+
+convertToCartesianShape
+required_capability: semantic_text_type
+
+FROM semantic_text METADATA _id
+| EVAL result = to_cartesianshape(st_cartesian_shape)
+| KEEP _id, result
+| SORT _id
+;
+
+_id:keyword | result:cartesian_shape
+1           | null
+2           | null
+3           | null
+;
+
+convertToDatetime
+required_capability: semantic_text_type
+
+FROM semantic_text METADATA _id
+| EVAL result = to_datetime(st_datetime)
+| KEEP _id, result
+| SORT _id, result
+;
+
+_id:keyword|result:datetime
+1          | 1953-09-02T00:00:00.000Z
+2          | 2023-09-24T15:57:00.000Z
+3          | null
+;
+
+convertToDouble
+required_capability: semantic_text_type
+
+FROM semantic_text METADATA _id
+| EVAL result = to_double(st_double)
+| KEEP _id, result
+| SORT _id
+;
+
+_id:keyword|result:double
+1          | 5.20128E11
+2          | 4541.11
+3          | null
+;
+
+convertToGeopoint
+required_capability: semantic_text_type
+
+FROM semantic_text METADATA _id
+| EVAL result = to_geopoint(st_geopoint)
+| KEEP _id, result
+| SORT _id
+;
+
+_id:keyword | result:geo_point
+1           | "POINT(42.97109630194 14.7552534413725)"
+2           | "POINT(37.97109630194 21.7552534413725)"
+3           | null
+;
+
+convertToGeoshape
+required_capability: semantic_text_type
+
+FROM semantic_text METADATA _id
+| EVAL result = to_geoshape(st_geoshape)
+| KEEP _id, result
+| SORT _id
+;
+
+_id:keyword | result:geo_shape
+1           | "POLYGON ((30 10, 40 40, 20 40, 10 20, 30 10))"
+2           | "POLYGON ((30 10, 40 40, 20 40, 10 20, 30 10))"
+3           | null
+;
+
+convertToInteger
+required_capability: semantic_text_type
+
+FROM semantic_text METADATA _id
+| EVAL result = to_integer(st_integer)
+| KEEP _id, result
+| SORT _id
+;
+
+_id:keyword | result:integer
+1           | 23
+2           | 122
+3           | null
+;
+
+convertToIp
+required_capability: semantic_text_type
+
+FROM semantic_text METADATA _id
+| EVAL result = to_ip(st_ip)
+| KEEP _id, result
+| SORT _id
+;
+
+_id:keyword | result:ip
+1           | 1.1.1.1
+2           | 1.1.2.1
+3           | null
+;
+
+convertToLong
+required_capability: semantic_text_type
+
+FROM semantic_text METADATA _id
+| EVAL result = to_long(st_long)
+| KEEP _id, result
+| SORT _id
+;
+
+_id:keyword | result:long
+1           | 2147483648
+2           | 123
+3           | null
+;
+
+convertToUnsignedLong
+required_capability: semantic_text_type
+
+FROM semantic_text METADATA _id
+| EVAL result = to_unsigned_long(st_unsigned_long)
+| KEEP _id, result
+| SORT _id
+;
+
+_id:keyword | result:unsigned_long
+1           | 2147483648
+2           | 2147483648.2
+3           | null
+;
+
+convertToVersion
+required_capability: semantic_text_type
+
+FROM semantic_text METADATA _id
+| EVAL result = to_version(st_version)
+| KEEP _id, result
+| SORT _id
+;
+
+_id:keyword | result:version
+1           | 1.2.3
+2           | 9.0.0
+3           | null
+;
+
+concat
```

Comment:
tests for [String functions](https://www.elastic.co/guide/en/elasticsearch/reference/master/esql-functions-operators.html#esql-string-functions)